popup.js

```diff
@@ -12,23 +12,23 @@ const AGENCY_IDS = {
 };
 
 function debounce(fn, duration) {
-  let timeout = null;
+  var timeout = null;
 
   return function () {
-    const args = arguments;
+    var args = arguments;
 
     if (timeout) {
       clearTimeout(timeout);
     }
 
-    timeout = setTimeout(() => {
+    timeout = setTimeout(function(){
       timeout = null;
       fn.apply(this, args);
```

Comment:
Because this is no longer an arrow function, `this` no longer points to the outer scope. You could assign `var scope = this;` right after `var args = arguments;` and then do: `fn.apply(scope, args)` inside the timeout handler.